plugin/bufferline.vim

```diff
@@ -205,6 +205,7 @@ endfunc
 "========================
 
 function! s:on_buffer_open(abuf)
+   lua require'bufferline.state'.update_names()
```

Comment:
I know I'm getting close to scope creep on this PR, but during my usage I experienced another issue wherein opening a tab and then entering a buffer from that fresh tab would not change the tabline. This adds a binding for updating names to whenever a `BufNewFile` event fires, and fixes the issue.

Reply:
:+1:

Reply:
I've reviewed the code further, and this code shouldn't actually be called at this point. The way it's supposed to work is that any update to the buffers state is done in `state.get_updated_buffers()`. If something isn't updating when it should, the missing autocommand should be added in `plugin/bufferline.vim`

Reply:
Got it, I'll move this. Thanks!

Reply:
@romgrk I'm a little confused by what you mean. The file this change _is_ to is `plugin/bufferline.vim`, on [the function](https://github.com/romgrk/barbar.nvim/blob/master/plugin/bufferline.vim#L12-L13) that is run whenever a `BufNewFile` event fires.

Reply:
Anytime the buffferline needs to be updated, `bufferline#update()` should be called. This function isn't supposed to be called this way, I'm not even sure why this works because the function in itself doesn't update the statusline. What I meant was I'd rather add an event [here](https://github.com/romgrk/barbar.nvim/blob/master/plugin/bufferline.vim#L27).

Reply:
Okay! I'll move it to the place that better suits it.

Reply:
I found out what's causing the issue. I'm glad you told me to keep looking, or I never would have found it! [This](https://github.com/romgrk/barbar.nvim/blob/master/lua/bufferline/state.lua#L238-L241) is the _real_ problem:

```lua
function m.get_updated_buffers()
  local current_buffers = vim.fn['bufferline#filter']('&buflisted')
  local new_buffers =
    filter(
      function(b) return not includes(m.buffers, b) end,
      current_buffers)

  -- …

  if len(new_buffers) > 0 then
    did_change = true

    open_buffers(new_buffers)
  end

  -- …

  if did_change then
    m.update_names()
  end
end
```

Both `m.buffers` and `b` are referring to buffer numbers, rather than buffer names. When a '[No name]' buffer is replaced with a named buffer, the buffer number remains the same but the buffer name changes. As such, there are no new buffers, but there are new buffer _names_.